### PR TITLE
fix(hardware-profiles): add character limits to name and description fields

### PR DESCRIPTION
**Pull request by kanishka-commits**
https://issues.redhat.com/browse/RHOAIENG-20550

## Description

This PR adds character limits to hardware profile name and description fields to prevent UI truncation issues when creating or editing hardware profiles.

**What changed:**
- Added character limit constants: 128 chars for display name, 255 chars for description
- Updated validation schema with `.max()` constraints using Zod
- Added `maxLength` and `maxLengthDesc` props to the `K8sNameDescriptionField` component
- Character count indicators will now appear when approaching the limit

**Why:**
Hardware profiles had no character limits, causing names and descriptions to truncate in the UI. The limits (128 for name, 255 for description) follow established patterns in the codebase (Model Registry uses 128, Pipelines use 255).

**Files modified:**
- `frontend/src/pages/hardwareProfiles/manage/const.ts` - Added character limit constants
- `frontend/src/pages/hardwareProfiles/manage/validationUtils.ts` - Added validation constraints
- `frontend/src/pages/hardwareProfiles/manage/ManageHardwareProfile.tsx` - Passed maxLength props to component
- `frontend/src/pages/hardwareProfiles/manage/__tests__/validationUtils.spec.ts` - Added comprehensive test coverage

## How Has This Been Tested?

**Automated tests:**
- Added 4 new unit tests for character limit validation
- Tests verify that names/descriptions exceeding limits fail validation
- Tests verify that names/descriptions at exact limits pass validation
- All tests use constants for maintainability

**Manual testing recommended:**
1. Create a new hardware profile with a name exactly 128 characters - should succeed
2. Attempt to create a hardware profile with a name > 128 characters - should show validation error
3. Create a hardware profile with description exactly 255 characters - should succeed
4. Attempt to create a hardware profile with description > 255 characters - should show validation error
5. Verify character count warning appears when approaching limit

## Test Impact

✅ Added comprehensive unit tests covering:
- Display name exceeding limit (129+ chars)
- Display name at exact limit (128 chars)
- Description exceeding limit (256+ chars)
- Description at exact limit (255 chars)

All tests use imported constants for better maintainability.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)
- [x] The code follows our [Best Practices](/docs/best-practices.md) (React coding standards, PatternFly usage, performance considerations)

If you have UI changes: 
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Improvements**
  * Hardware profile display names and descriptions now enforce character limits (128 and 255 characters respectively).
  * Description field helper text now warns earlier for smaller limits while keeping a generous buffer for larger limits.

* **Tests**
  * Added test coverage validating character-limit boundary conditions for display name and description.

### CODE REVIEW

**Review by kanishka-commits — COMMENTED** [review]
## Tested and Verified: 
*Before*
Image

Image

___

*After*

Image

File: frontend/src/pages/hardwareProfiles/manage/ManageHardwareProfile.tsx [hunk 1]
```diff
@@ -157,6 +161,8 @@ const ManageHardwareProfile: React.FC<ManageHardwareProfileProps> = ({
                 data={profileNameDesc}
                 onDataChange={setProfileNameDesc}
                 dataTestId="hardware-profile-name-desc"
+                maxLength={HARDWARE_PROFILE_DISPLAY_NAME_CHAR_LIMIT}
```
Comment (manaswinidas) [inline frontend/src/pages/hardwareProfiles/manage/ManageHardwareProfile.tsx:164]: Nit: With `maxLengthDesc=255`, the `K8sNameDescriptionField` component will show the "Cannot exceed 255 characters" warning once the description exceeds **5 characters** — because its `showDescWarning` threshold is hardcoded as `maxLengthDesc - 250`. That's essentially always-on.

This is a pre-existing issue in the shared component, but this PR is the first to expose it with a small limit. Consider either:
- Omitting `maxLengthDesc` here (the Zod `.max()` still validates on submit), or
- Fixing the threshold in `K8sNameDescriptionField` to use a proportional offset (e.g. `Math.max(0, maxLengthDesc - Math.min(250, maxLengthDesc * 0.1))`).

**Review by manaswinidas — COMMENTED** [review]
Looks good overall — small, focused change that follows existing patterns. One UX concern worth addressing:

File: frontend/src/pages/hardwareProfiles/manage/__tests__/validationUtils.spec.ts [hunk 2]
```diff
@@ -67,6 +71,56 @@ describe('manageHardwareProfileValidationSchema', () => {
...
+  it('should pass when description is exactly at character limit', () => {
+    const validDataWithMaxDescription = {
+      ...validData,
+      description: 'a'.repeat(HARDWARE_PROFILE_DESCRIPTION_CHAR_LIMIT),
+    };
+
+    const result = manageHardwareProfileValidationSchema.safeParse(validDataWithMaxDescription);
+    expect(result.success).toBe(true);
+  });
+
```
Comment (manaswinidas) [inline frontend/src/pages/hardwareProfiles/manage/__tests__/validationUtils.spec.ts]: Nit: These four tests follow two repeating templates (exceed → fail, exact → pass) across two fields. Consider collapsing with `it.each`:

```ts
it.each([
  ['displayName', HARDWARE_PROFILE_DISPLAY_NAME_CHAR_LIMIT, 'Display name'],
  ['description', HARDWARE_PROFILE_DESCRIPTION_CHAR_LIMIT, 'Description'],
])('should fail when %s exceeds character limit', (field, limit, label) => {
  const result = manageHardwareProfileValidationSchema.safeParse({
    ...validData,
    [field]: 'a'.repeat(limit + 1),
  });
  expect(result.success).toBe(false);
});
```

Also, the error message assertions (`toContain`) couple the tests to exact wording — if the message text changes the tests break even though validation still works. `expect(result.success).toBe(false)` is sufficient here; the message check is optional.

File: frontend/src/pages/hardwareProfiles/manage/ManageHardwareProfile.tsx (hunk 1 above)
Comment (kanishka-commits) [reply frontend/src/pages/hardwareProfiles/manage/ManageHardwareProfile.tsx:164]: Thanks for pointing out! Updated `K8sNameDescriptionField` to replace the hardcoded `-250` threshold with a proportional offset.

**Logic:**
```ts
description.length > maxLengthDesc - Math.min(250, Math.floor(maxLengthDesc * 0.1))
```

- maxLengthDesc = 255 → warning at 230 chars (10%)
- maxLengthDesc = 5500 → unchanged (5250 chars)

*before*
Screenshot 2026-04-21 at 12 40 49 PM

*after*
Screenshot 2026-04-21 at 12 41 16 PM

File: frontend/src/pages/hardwareProfiles/manage/__tests__/validationUtils.spec.ts (hunk 2 above)
Comment (kanishka-commits) [reply frontend/src/pages/hardwareProfiles/manage/__tests__/validationUtils.spec.ts]: Makes sense, collapsed the four tests into two `it.each` blocks and dropped the `toContain` message assertions. Tests now just verify `result.success` without coupling to exact wording.

**Review by manaswinidas — APPROVED** [review]
/lgtm
/approve
